src/main/java/persistence/entity/EntityLoaderImpl.java

```diff
@@ -10,20 +14,66 @@
 public class EntityLoaderImpl implements EntityLoader {
 
     private final JdbcTemplate jdbcTemplate;
-    private final Dialect dialect;
     private final SelectQueryBuilder selectQueryBuilder;
 
 
-    public EntityLoaderImpl(JdbcTemplate jdbcTemplate, Dialect dialect) {
+    public EntityLoaderImpl(JdbcTemplate jdbcTemplate) {
         this.jdbcTemplate = jdbcTemplate;
-        this.dialect = dialect;
         this.selectQueryBuilder = new SelectQueryBuilder();
     }
 
+    public <T> List<T> findAll(Class<T> clazz) {
+        SelectQueryBuilder queryBuilder = selectQueryBuilder.build(clazz);
+        TableColumn tableColumn = new TableColumn(clazz);
+        JoinTableColumns joinTableColumns = tableColumn.getJoinTableColumns();
+        if (joinTableColumns.hasNotAssociation()) {
+            String query = queryBuilder.selectFromClause();
+            return jdbcTemplate.query(query, new GenericRowMapper<>(clazz));
+        }
+        String query = queryBuilder.selectFromJoinClause(joinTableColumns);
+        return jdbcTemplate.query(query, new GenericRowMapper<>(clazz));
+    }
+
     @Override
     public <T> T find(Class<T> clazz, Long id) {
         SelectQueryBuilder queryBuilder = selectQueryBuilder.build(clazz);
-        String query = queryBuilder.toStatementWithId(id);
-        return jdbcTemplate.queryForObject(query, new GenericRowMapper<>(clazz));
+        TableColumn tableColumn = new TableColumn(clazz);
+        JoinTableColumns joinTableColumns = tableColumn.getJoinTableColumns();
+        if (joinTableColumns.hasNotAssociation()) {
+            String query = queryBuilder.selectFromWhereIdClause(id);
+            return jdbcTemplate.queryForObject(query, new GenericRowMapper<>(clazz));
+        }
+        T instance = getInstance(clazz, id, joinTableColumns);
+
+        List<JoinTableColumn> lazyJoinTables = joinTableColumns.getLazyJoinTables();
+        setProxy(lazyJoinTables, instance);
+        return instance;
+    }
+
+    private <T> T getInstance(Class<T> clazz, Long id, JoinTableColumns joinTableColumns) {
+        SelectQueryBuilder queryBuilder = selectQueryBuilder.build(clazz);
+        if (joinTableColumns.hasEager()) {
+            JoinTableColumns eagerJoinTables = joinTableColumns.getEagerJoinTables();
+            String query = queryBuilder.selectFromJoinWhereIdClause(eagerJoinTables, id);
+            return jdbcTemplate.queryForObject(query, new GenericRowMapper<>(clazz));
+        }
+
+        String fromQuery = queryBuilder.selectFromWhereIdClause(id);
+        return jdbcTemplate.queryForObject(fromQuery, new GenericRowMapper<>(clazz));
+    }
```

Comment:
```suggestion
        return getInstance(clazz, id, joinTableColumns);
    }

    private <T> T getInstance(Class<T> clazz, Long id, JoinTableColumns joinTableColumns) {
        SelectQueryBuilder queryBuilder = selectQueryBuilder.build(clazz);
        if (joinTableColumns.hasEager()) {
            JoinTableColumns eagerJoinTables = joinTableColumns.getEagerJoinTables();
            String query = queryBuilder.selectFromJoinWhereIdClause(eagerJoinTables, id);
            return jdbcTemplate.queryForObject(query, new GenericRowMapper<>(clazz));
        }

        String fromQuery = queryBuilder.selectFromWhereIdClause(id);
        T instance = jdbcTemplate.queryForObject(fromQuery, new GenericRowMapper<>(clazz));

        if (joinTableColumns.hasLazy()) {
            List<JoinTableColumn> lazyJoinTables = joinTableColumns.getLazyJoinTables();
            setProxy(lazyJoinTables, instance);
        }

        return instance;
    }
```
사소한 코멘트인데 조금더 응집성이 올라갈 수 있지 않을까 싶어서 제안드려봅니다
hasLazy, hasEager 상태는 는 일급컬렉션 생성시 미리 만들어 놓아도, 아니어도 어느쪽이든 좋을것 같습니다 :)

Reply:
오.. 저는 한 메서드에 if문이 2~3개 있으면 분리하는 편이라서 if문을 기준으로 분리를 진행했었는데요 제안 주신 방식도 잘 읽혀서 좋은것 같네요! 반영했습니다:)